slack-lint/src/test/java/slack/lint/text/SpanPointMarkDangerousCheckDetectorTest.kt

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright (C) 2021 Slack Technologies, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package slack.lint.text
+
+import org.intellij.lang.annotations.Language
+import org.junit.Test
+import slack.lint.BaseSlackLintTest
+
+/**
+ * Tests [SpanPointMarkDangerousCheckDetector].
+ */
+class SpanPointMarkDangerousCheckDetectorTest : BaseSlackLintTest() {
+
+  override fun getDetector() = SpanPointMarkDangerousCheckDetector()
+  override fun getIssues() = listOf(SpanPointMarkDangerousCheckDetector.ISSUE)
+
+  private val androidTextStubs = kotlin(
+    """
+        package android.text
+
+        object Spanned {
+            const val SPAN_INCLUSIVE_INCLUSIVE = 1
+            const val SPAN_INCLUSIVE_EXCLUSIVE = 2
+            const val SPAN_EXCLUSIVE_INCLUSIVE = 3
+            const val SPAN_EXCLUSIVE_EXCLUSIVE = 4
+            const val SPAN_POINT_MARK_MASK = 0xFF
+        }
+      """
+  ).indented()
+
+  @Test
+  fun `visitBinaryExpression - given conforming expression - has clean report`() {
+    @Language("kotlin")
+    val testFile = kotlin(
+      """
+                package slack.text
+
+              import android.text.Spanned
+
+              class MyClass {
+                  fun doCheckCorrectly(spanned: Spanned): Boolean {
+                    return spanned.getSpanFlags(Object()) and Spanned.SPAN_POINT_MARK_MASK == Spanned.SPAN_INCLUSIVE_INCLUSIVE
+                  }
+              }
+            """
+    ).indented()
+    lint()
+      .files(androidTextStubs, testFile)
+      .issues(SpanPointMarkDangerousCheckDetector.ISSUE)
+      .run()
+      .expectClean()
+  }
+
+  @Test
+  fun `visitBinaryExpression - given violating expression with INCLUSIVE_INCLUSIVE on left - creates error and fix`() {
```

Comment:
Same here with redundant "given violating expression with...". Let's keep these names simple, for example

`error - INCLUSIVE_INCLUSIVE on left`

Seeing a list of these in test results or the IDE functions gutter is much more readable at a glance. I know there's a push for GWT here, but I think it's a little overkill in this case.

Reply:
GWT?

Reply:
"Given When Then". Maybe not the exact pattern you're going for here but quite similar and it's just a bit unnecessarily verbose